morf-h2/src/main/java/org/alfasoftware/morf/jdbc/h2/H2Dialect.java

```diff
@@ -567,77 +576,48 @@ protected String getSqlForAddMonths(Function function) {
 
 
   /**
-   *  TODO
-   * The following is a workaround to a bug in H2 version 1.4.200 whereby the MERGE...USING statement does not release the source select statement
-   * Please remove this method once https://github.com/h2database/h2database/issues/2196 has been fixed and H2 upgraded to the fixed version
-   * This workaround uses the following alternative syntax, which fortunately does not lead to the same bug:
-   *
-   * <pre>
-   *   WITH xmergesource AS (SELECT ...)
-   *   MERGE INTO Table
-   *     USING xmergesource
-   *     ON (Table.id = xmergesource.id)
-   *     WHEN MATCHED THEN UPDATE ...
-   *     WHEN NOT MATCHED THEN INSERT ...
-   * </pre>
-   *
    * @see org.alfasoftware.morf.jdbc.SqlDialect#getSqlFrom(org.alfasoftware.morf.sql.MergeStatement)
    */
   @Override
   protected String getSqlFrom(MergeStatement statement) {

```

Comment:
This new Merge statement version does not work with statement.getIfUpdating() does it?
TestAccumulatingMergeStatement should fail if it does not